include/nighthawk/common/uri.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+#include "common/network/dns_impl.h"
+#include "common/network/utility.h"
+
+#include "nighthawk/common/exception.h"
+
+namespace Nighthawk {
+
+/**
+ * Any exception thrown by Uri shall inherit from UriException.
+ */
+class UriException : public NighthawkException {
+public:
+  UriException(const std::string& message) : NighthawkException(message) {}
+};
+
+/**
+ * Abstract Uri interface.
+ */
+class Uri {
+public:
+  virtual ~Uri() = default;
+
+  /**
+   * @return absl::string_view containing the "host:port" fragment of the parsed uri. The port
+   * will be explicitly set in case it is the default for the protocol. Do not use the returned
+   * value after the Uri instance is destructed.
+   */
+  virtual absl::string_view host_and_port() const PURE;
+
+  /**
+   * @return absl::string_view containing the "host" fragment parsed uri. Do not use the
+   * returned value after the Uri instance is destructed.
+   */
+  virtual absl::string_view host_without_port() const PURE;
```

Comment:
Tiny nit: I'd prefer `host()` and `hostWithport()` here. But, if CI passes, I'm happy to merge and have this combined in some later PR, it's not that important.

Reply:
done